Prox/Prox/PlaceCarousel/PlaceCarouselViewController.swift

```diff
@@ -360,3 +358,26 @@ extension PlaceCarouselViewController: LocationProvider {
     }
 }
 
+extension PlaceCarouselViewController: PlacesControllerDelegate {
+    func placesControllerNoPlacesFound(_ controller: PlacesController, willRetry retry: Bool) {
+        if retry {
+            // TODO placeholder for the waiting state.
+            headerView.numberOfPlacesLabel.text = "Waiting"
+        } else {
+            self.places = []
+        }
+
+    }
+
+    func placesController(_ controller: PlacesController, didUpdatePlaces places: [Place]) {
+        self.places = places
```

Comment:
The `refresh` is redundant (see above).

I've put the inequality test in the setter, to make it idempotent.